lib/ddtrace/contrib/sinatra/framework.rb

```diff
@@ -0,0 +1,77 @@
+module Datadog
+  module Contrib
+    # Instrument Sinatra.
+    module Sinatra
+      # Sinatra framework code, used to essentially:
+      # - handle configuration entries which are specific to Datadog tracing
+      # - instrument parts of the framework when needed
+      module Framework
+        # Configure Rack from Sinatra, but only if Rack has not been configured manually beforehand
+        def self.setup
+          return if Datadog::Tracing.configuration.instrumented_integrations.key?(:rack)
```

Comment:
Why is this guard necessary? When does it run multiple times? I would think if it ran configuration twice this shouldn't be an issue. I can see this being especially problematic for tests where it generates and re-applies patches between examples and monkey-patches hang around (in which case it won't run the configure step.)

Reply:
I was presuming that if Rack was configured manually by the user we might want to keep that configuration instead of reconfiguring it, possibly overrwriting/resetting the user provided configuration.

Reply:
Removed for now, for consistency with the Rails integration.